src/core/server/ui_settings/saved_objects/permission_controlled_ui_settings_wrapper.ts

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ACL,
+  SavedObject,
+  SavedObjectsBaseOptions,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsClientWrapperOptions,
+  SavedObjectsCreateOptions,
+  SavedObjectsErrorHelpers,
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+} from 'opensearch-dashboards/server';
+import { getWorkspaceState } from 'opensearch-dashboards/server/utils';
+import { adminUiSettings } from './admin_ui_settings';
+import { DASHBOARD_ADMIN_SETTINGS_ID } from '../utils';
+
+type AttributesObject = Record<string, unknown>;
+
+/**
+ * Wrapper for admin UI settings that enforces permission controls
+ * Handles special cases for admin UI settings with appropriate access controls
+ */
+export class PermissionControlledUiSettingsWrapper {
+  private aclInstance?: ACL;
+
+  /**
+   * @param isPermissionControlEnabled
+   */
+  constructor(private readonly isPermissionControlEnabled: boolean) {}
+
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const getUiSettingsWithPermission = async <T = unknown>(
```

Comment:
Do we need `getUiSettingsWithPermission` method? Seems we can simply use `wrapperOptions.client.get`.

Reply:
Right, I previously think we need to manual handle 404 case but seems this can be done by `ignore404Errors`, will remove `getUiSettingsWithPermission` then.